core/src/test/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationServiceSpec.scala

```diff
@@ -526,4 +490,77 @@ class GoogleProjectRegistrationServiceSpec
+    val mockGoogleProjectRegRepo = mock[GoogleProjectRegistrationRepository]
+    val mockContext = mock[RawlsRequestContext]
+    val mockBillingRepository = mock[BillingRepository]
+    val mockGoogleServicesDAO = mock[GoogleServicesDAO]
+
+    when(mockGoogleServicesDAO.setBillingAccountName(any[GoogleProjectId], any[RawlsBillingAccountName], any()))
+      .thenReturn(
+        Future.failed(
+          new RawlsExceptionWithErrorReport(errorReport =
+            ErrorReport(StatusCodes.Forbidden, "Something has gone wrong in Google")
+          )
+        )
+      )
+
+    val billingProjectId = RawlsBillingProjectName("billing-project-id")
+
+    val billingProject: RawlsBillingProject = RawlsBillingProject(UUID.randomUUID(),
+                                                                  billingProjectId,
+                                                                  CreationStatuses.Ready,
+                                                                  Option(RawlsBillingAccountName("billing-account")),
+                                                                  None
+    )
+
+    when(mockBillingRepository.getBillingProject(mockitoEq(billingProjectId)))
+      .thenReturn(Future.successful(Some(billingProject)))
+
+    when(
+      mockSamDAO.userHasAction(any[SamResourceTypeName], any[String], any[SamResourceAction], any[RawlsRequestContext])
+    ).thenReturn(Future.successful(true))
+
+    when(
+      mockSamDAO.listUserActionsForResource(mockitoEq(SamResourceTypeNames.billingProject),
+                                            any[String],
+                                            any[RawlsRequestContext]
+      )
+    ).thenReturn(
+      Future.successful(Set(SamBillingProjectActions.link, SamBillingProjectActions.own))
+    )
+
+    val googleProjectRegService =
+      GoogleProjectRegistrationService.constructor(
+        mockSamDAO,
+        mockGoogleProjectRegRepo,
+        mockBillingRepository,
+        mockGoogleServicesDAO
+      )(
+        mockContext
+      )
+
+    when(mockGoogleProjectRegRepo.deleteGoogleProjectRegistration(any[GoogleProjectId]))
+      .thenReturn(Future.successful(true))
+
+    val testProject = GoogleProjectRegistration(GoogleProjectId("test-project"),
+                                                Some(RawlsBillingAccountName("billing-account")),
+                                                None,
+                                                billingProjectId
+    )
+    when(mockGoogleProjectRegRepo.registerGoogleProject(any[GoogleProjectRegistration])(any[TestExecutionContext]))
+      .thenReturn(Future.successful(Some(testProject)))
+
+    val e = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        googleProjectRegService.registerGoogleProject(testProject),
+        Duration.Inf
+      )
+    }
+
+    e.errorReport.message shouldBe "Failed to set billing account in Google: Something has gone wrong in Google"
+
```

Comment:
It would be good to verify that `deleteGoogleProjectRegistration` was called as well